core/event/task/task.go

```diff
@@ -37,6 +37,17 @@ func Once(task Task) Task {
 	}
 }
 
+// Wraps a task in a coroutine to asynchronously execute after a specified duration
```

Comment:
The godoc stuff requires a very specific format for documentation comments. Comments for functions need to start with the function name. So this needs to be `// Delay ...`.
Please see https://blog.golang.org/godoc-documenting-go-code for details.